models/src/main/scala/coop/rchain/models/rholang/implicits.scala

```diff
@@ -404,7 +404,9 @@ object implicits {
   implicit val ReceiveBindLocallyFree: HasLocallyFree[ReceiveBind] =
     new HasLocallyFree[ReceiveBind] {
       def connectiveUsed(rb: ReceiveBind) =
-        ChannelLocallyFree.connectiveUsed(rb.source.get)
+        ChannelLocallyFree.connectiveUsed(rb.source.get) || rb.patterns.exists(
+          ChannelLocallyFree.connectiveUsed(_))
```

Comment:
I am not yet sure about this change, reverting to previous version does not make tests fail but newer versions seems correct.

Reply:
This isn't correct. The patterns may have connectives without the ReceiveBind being a pattern itself.